test/tests/ha/rackdh_ha_resource_test.py

```diff
@@ -0,0 +1,120 @@
+"""
+Copyright 2017, Dell, Inc.
+
+This test checks the installation of the HA environment
+"""
+
+import fit_common
+
+# LOCAL
+from nose.plugins.attrib import attr
+from nosedep import depends
+
+FAIL_ON_NODE_COLLECTION = True
+active_node_list = list()
+
+
+@attr(regression=False, smoke=False, ha_resource_tests=True)
+class HaResourceTest(fit_common.unittest.TestCase):
+
+    def setUp(self):
+        self.number_of_vms = int(fit_common.fitargs()['numvms'])
+        self.number_of_services = self.number_of_vms - 1  # number of mongo and rabbitmq resource
+
+    def is_service_up(self, service, vmnum):
+        command = 'service {} status | grep -q "running"'.format(service)
+        rc = fit_common.remote_shell(command, vmnum=vmnum)['exitcode']
+        return rc == 0
+
+    def has_online_nodes(self, vmnum, node_name):
+        command = 'crm status | grep Online | grep {}'.format(node_name)
+        status = fit_common.remote_shell(command, vmnum=vmnum)['exitcode']
+        return status == 0
+
+    def has_resource(self, vmnum, resource):
+        command = "crm resource show | grep '{}'".format(resource)
+        rc = fit_common.remote_shell(command, vmnum=vmnum)['exitcode']
+        return rc == 0
+
+    def test010_check_pacemaker_status(self):
+        for vmnum in range(1, self.number_of_vms + 1):
+            # Check that corosync service is active and running
+            self.assertTrue(self.is_service_up("corosync", vmnum), "Corosync not running on node {}".format(vmnum))
+            # Check that pacemaker service is active and running
+            self.assertTrue(self.is_service_up("pacemaker", vmnum), "Pacemaker not running on node {}".format(vmnum))
+
+    def test011_check_default_configuration(self):
+        for vmnum in range(1, self.number_of_vms + 1):
+            # Check for default configuration
+            self.assertFalse(self.has_online_nodes(vmnum, 'ora'),
+                             "Default configuration on vm {}".format(vmnum))
+
+    def test012_check_node_status(self):
+        for vmnum in range(1, self.number_of_vms + 1):
+            for node_number in range(1, self.number_of_vms + 1):
+                # Determine if this node is online
+                self.assertTrue(self.has_online_nodes(vmnum, 'node{}'.format(node_number)),

```

Comment:
Do you want to exit the test on the first failure here, or do you want to check each vmnum/node combo an report back a list of failures?
if so, I suggest creating an errorlist = [], append the error to the errorlist.
When done, check if errorlist is not empty, dump the list of nodes/errors that are not online and then assert.
Then you can see what is failing.